src/contracts/InvestorChef.sol

```diff
@@ -0,0 +1,24 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./TopChef.sol";
+
+contract InvestorChef is TopChef {
+    address public groupAddress;
+    uint256 public agIndex;
+    uint256 public shares;
+    bool public newAutoDistribute;
+
+    constructor(address metricTokenAddress) TopChef(metricTokenAddress) {
+        setMetricToken(metricTokenAddress);
```

Comment:
wait doesn't this call every function from the constructor?

Reply:
Yeah which I thought was odd but took your message too literally ha ha. What is the solidity pattern? 
```
function methodContainer {
          parentMethod1(),
          parentMethod2(),
    }
 ```
 or
 ```
 function method1 {
          parentMethod1()
     }
 function method2 {
           parentMethod2()
    }
 ```

Reply:
no don't need anything!  it should just inherit em

`contract InvestorChef is TopChef {
    constructor(address metricTokenAddress) TopChef(metricTokenAddress) {
}
}
`